bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/regexp/TreeBuilder.java

```diff
@@ -182,6 +243,11 @@ private String readRegUnicodeScript() {
         throw new BallerinaException(getErrorMsg(nextToken));
     }
 
+    /**
+     * Read the unicode general category.
+     *
+     * @return gc=category or category
```

Comment:
Is this dock comment correct?

Reply:
This is correct. Lets say we are parsing pattern like `\p{gc=L}` this will return `gc=L`. And if we are parsing a pattern like `\p{L} this will return `L`.